src/odemis/acq/stream/_base.py

```diff
@@ -787,6 +835,172 @@ def _image_thread(wstream):
 
         gc.collect()
 
+    def _zFromMpp(self):
+        """
+        Return the zoom level based on the current .mpp value
+        return (int): The zoom level based on the current .mpp value
+        """
+        md = self.raw.metadata
+        ps = md[model.MD_PIXEL_SIZE]
+        return int(math.log(self.mpp.value / ps[0], 2))
+
+    def _rectWorldToPixel(self, rect):
+        """
+        Convert rect from world coordinates to pixel coordinates
+        rect (tuple containing x1, y1, x2, y2): Rect on world coordinates
+        return (tuple containing x1, y1, x2, y2): Rect on pixel coordinates
+        """
+        md = self.raw.metadata
+        ps = md.get(model.MD_PIXEL_SIZE, (1e-6, 1e-6))
+        pos = md.get(model.MD_POS, (0.0, 0.0))
+        # Removes the center coordinates of the image. After that, rect will be centered on 0, 0
+        rect = (
+            rect[0] - pos[0],
+            rect[1] - pos[1],
+            rect[2] - pos[0],
+            rect[3] - pos[1]
+        )
+        dims = md.get(model.MD_DIMS, "CTZYX"[-self.raw.ndim::])
+        img_shape = (self.raw.shape[dims.index('X')], self.raw.shape[dims.index('Y')])
+
+        return (
+            int(rect[0] / ps[0] + img_shape[0] / 2),
+            int(rect[1] / ps[1] + img_shape[1] / 2),
+            int(rect[2] / ps[0] + img_shape[0] / 2),
+            int(rect[3] / ps[1] + img_shape[1] / 2),
+        )
+
+    def getMergedImage(self):
```

Comment:
No, do not provide such a method! It is equivalent to `util.img.mergeTiles(s.image.value)`. => replace all the callers by such a code.